### DISCUSSION TITLE
CMYK PDF conversion colors issue

### DISCUSSION THREAD

**Discussion by lde-net2** [dlemstra/Magick.NET#1406] [Help] [answered]
Hello,

Lately we are facing some issues with images produced out of some, not all, cmyk PDFs.
I updated to the latest magick.net version (13.1.3) and GS as well (10.01.2)

The jpeg output show some color shifts, and some artifacts
while the direct output from from ghostscript command line is correct

exemples
![image](https://github.com/dlemstra/Magick.NET/assets/127402688/3226394e-e9e5-464b-8901-4d261ff4e137)
![image](https://github.com/dlemstra/Magick.NET/assets/127402688/ec84eb27-7aa8-4dbe-bc9e-649eebba1d12)

i checked the PDFs with acrobat pro preflight, and all PDF resources are in CMYK 
and they are in the same PDF version (1.3) than the non offending PDF

i downgraded the GS libs to an older 9.50 (2019), and miracle, no issue converting the very same PDFs,
with the exact same magick version & code !!
How come ?

not sure where to lookup more
thanks to help,

Luc

**Comment by lde-net2** [comment, marked as answer]
Got it.. breaking changes in GS 10.01+
https://github.com/ArtifexSoftware/ghostpdl-downloads/releases

"This release officially deprecates the old Postscript implementation of PDF, we will not be updating or maintaining that code moving forward. The option to use the old PDF implementation will be removed in the next full release (10.01.0)"

so i guess magick.net needs an update.

i downgraded GS to 10.0.0 with latest magick.NET => all good

**Comment by dlemstra** [comment]
Can you share a PDF file? This library has support for new version of Ghostscript and should work with that version. All the unit tests are also running with that. It would be nice if you could create a PDF with just a single square with a specific color that can be used to reproduce this.

**Comment by lde-net2** [comment]
Sure, here is one
[6430174f932c3363f415b3a0.pdf](https://github.com/dlemstra/Magick.NET/files/11897205/6430174f932c3363f415b3a0.pdf)

it was created by the studio, using indesign

**Comment by dlemstra** [comment]
How did you create the image on the right side? Which Ghostscript options did you use to create the image? I suspect that this is a bug inside Ghostscript but it might be possible that we are using the wrong command line options when converting the image.

**Comment by lde-net2** [comment]
"C:\Program Files\gs\gs10.01.2\bin\gswin64.exe" -sDEVICE=jpegcmyk -o out.jpg 6430174f932c3363f415b3a0.pdf

**Comment by lde-net2** [comment]
definitely a compat issue.
i reproduced it right away 
same code, same app, just changed the GS dlls in the Magick lib folder, nothing else
left : GS 10.01.2 => NOK
right : GS 10.0.0 => OK

![image](https://github.com/dlemstra/Magick.NET/assets/127402688/dcec74f7-f6b9-4d2f-9d42-a7838c0c833a)

i can't share my app code here, but i can demo you on call should you need.

**Comment by dlemstra** [comment]
We are using the `pamcmyk32` device and I suspect that this device now contains a bug? I think you will need to file a bug report there?

**Comment by lde-net2** [comment]
I guess i should be able to reproduce locally with the command line ?

this command does not output a valid file
`"C:\Program Files\gs\gs10.01.2\bin\gswin64.exe" -sDEVICE=pamcmyk32 -verbose -o out.jpg 6430174f932c3363f415b3a0.pdf`

what should i use to reproduce using your pamcmyk32 ?
thanks

**Reply by dlemstra** [reply]
That is a `out.pam` file instead of a jpg file. Not sure which viewer you can use to view that though.

**Comment by lde-net2** [comment]
https://filext.com/file-extension/PAM  but it cannot convert the generated file
